plugins/field-time/src/index.js

```diff
@@ -0,0 +1,120 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Time input field.
+ * @author carl.shen1@uwaterloo.ca (Carl Shen)
+ */
+
+import * as Blockly from 'blockly/core';
+
+/**
+ * Ensures that the input value is a valid time.
+ * @param {*=} newValue The input value.
+ * @return {?string} A valid time, or null if invalid.
+ */
+function timeValidator(newValue) {
+  if (!newValue) {
+    return null;
+  }
+  if (!/^([0-1]?[0-9]|2[0-3]):([0-5][0-9])$/.test(newValue)) {
+    return null;
+  }
+  return newValue;
+}
+
+/**
+ * Class for a time input field.
+ * @extends {Blockly.FieldTextInput}
+ */
+export class FieldTime extends Blockly.FieldTextInput {
+  /**
+   * Class for a time input field. Built using <input type='time'/> time picker
+   * @param {string=} value The initial value of the field. Should be in
+   *    '24-hour HH:MM' format. Defaults to 12:00.
+   * @param {Function=} validator A function that is called to validate
+   *    changes to the field's value. Takes in a time string & returns a
+   *    validated time string ('HH:MM' format), or null to abort the
+   *    change.
+   */
+  constructor(value = '12:00', validator = undefined) {
+    const initial = timeValidator(value) || '12:00';
+    super(initial, validator);
```

Comment:
Originally I just called `super('12:00', validator)` followed by `this.setValue(value)`, which seemed to work fine. However, the mocha test wouldn't let me do this (can't call setValue in the constructor?) so I factored out the validator.

Mocha error:
![image](https://user-images.githubusercontent.com/42399554/205412781-b6e16e4e-4e92-4141-8339-48b2028fdd7b.png)